roles/sudoers/tasks/teardown.yml

```diff
@@ -0,0 +1,32 @@
+---
+
+# Copyright 2021 Cloudera, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Remove the FreeIPA sudo rule
+- name: Remove sudo rule for passwordless sudo
+  community.general.ipa_sudorule:
+    ipa_user: "{{ sudoers__env_admin_username }}"
```

Comment:
`sudoers__env_admin_username` is not defined in `teardown.yml`. I suggest you pull out the username discovery into a shared task list like `initialize.yml` and invoke in both `main.yml` and `teardown.yml`.